beta/process.py

```diff
@@ -23,90 +24,122 @@
+            try:
+                stored_frame = self._lattice.get_center([pos_info["transform"]])['raw'].squeeze()
+            except:
+                continue
+
+            resname = pos_info["residue"].decode("utf-8")
+            pos_grp = chemical.functional_groups[resname]
+
+            dummy_pose.replace_residue(1, ResidueFactory.create_residue(res_type_set.name_map(resname)), False)
+
+            coords = [np.array([*dummy_pose.residues[1].xyz(atom)]) for atom in pos_grp.atoms]
+            c = np.stack(coords)
+
+            pos_frame = geometry.coords_to_transform(c)
+            final = np.dot(np.dot(ray_frame, stored_frame), np.linalg.inv(pos_frame))
+            dummy_pose.apply_transform(final)
+            yield dummy_pose.clone()
+
+    def search(self, pose, groups=["bidentate"], selector = TrueResidueSelector()):
+        pairs_of_rays = []
+        if np.any([x in groups for x in ["sc_sc", "bidentate"]]):
+            pairs_of_rays += look_for_sc_sc_bidentates(p, selector)
+        if np.any([x in groups for x in ["sc_scbb", "bidentate"]]):
+            pairs_of_rays += look_for_sc_scbb_bidentates(p, selector)
+        if np.any([x in groups for x in ["sc_bb", "bidentate"]]):
+            pairs_of_rays += look_for_sc_bb_bidentates(p, selector)
+
+        if np.any([x in groups for x in ["acceptor_acceptor", "network"]]):
+            pairs_of_rays += look_up_connected_network(p, selector, accA=True, accB=True)
+        if np.any([x in groups for x in ["acceptor_donor", "network"]]):
+            pairs_of_rays += look_up_connected_network(p, selector, accA=True, donB=True)
+        if np.any([x in groups for x in ["donor_acceptor", "network"]]):
+            pairs_of_rays += look_up_connected_network(p, selector, donA=True, accB=True)
+        if np.any([x in groups for x in ["donor_donor", "network"]]):
+            pairs_of_rays += look_up_connected_network(p, selector, donA=True, donB=True)
+
+        for (r1, r2) in pairs_of_rays:
+            yield from self.match(r1, r2, groups)
+
+    def filter_clash_minimize(p, hits, clash_cutoff=35.0, sfx=None, mmap=None):
+        if (not sfx):
+            sfx = scoring.ScoreFunctionFactory.create_score_function("beta_nov16")
+
+            sfx.set_weight(scoring.hbond_bb_sc, 2.0)
+            sfx.set_weight(scoring.hbond_sc, 2.0)
+
+        if (not mmap):
+            mmap = pyrosetta.rosetta.core.kinematics.MoveMap()
+
+            mmap.set_bb(False)
+            mmap.set_chi(False)
+            mmap.set_jump(1, True)
+
+        minmov = pyrosetta.rosetta.protocols.minimization_packing.MinMover(mmap, sfx, "dfpmin_armijo_nonmonotone", 0.01, False)
+
+        for hit in hits:
+            proto_pose = p.clone()
+            proto_pose.append_pose_by_jump(hit, len(proto_pose.residues))
+
+            sfx(proto_pose)
+
+            fa_rep = proto_pose.energies().total_energies()[scoring.fa_rep]
+            fa_elec = proto_pose.energies().total_energies()[scoring.fa_elec]
```

Comment:
Is this too strict?

Reply:
Possibly, but 35 kcal is a pretty big number. I suspect that testing on some real-world examples will be the best way to answer that question.